### PR TITLE
Java: Add command `Touch`. (Generic Command Group)

### PR DESCRIPTION
**Pull request by SanHalacogluImproving**

### CODE REVIEW

File: java/client/src/main/java/glide/api/commands/GenericBaseCommands.java [hunk 1]
```diff
@@ -407,4 +407,19 @@ CompletableFuture<Boolean> pexpireAt(
      * }</pre>
      */
     CompletableFuture<Long> objectRefcount(String key);
+
+    /**
+     * Updates the last access time of specified <code>keys</code> and returns the count of keys that
+     * were updated.
+     *
```
Comment (Yury-Fridlyand) [inline java/client/src/main/java/glide/api/commands/GenericBaseCommands.java:414]: Add a multi key note https://github.com/Bit-Quill/glide-for-redis/pull/226/files#diff-46c31f91e6bb4a1d2c763cddafd17d4aaad1b24f91744b604d6f11a26da9d45dR21-R22

File: java/integTest/src/test/java/glide/SharedCommandTests.java [hunk 2]
```diff
@@ -2769,4 +2769,20 @@ public void objectRefcount(BaseClient client) {
...
+        String key2 = "{key}-2" + UUID.randomUUID();
+        String key3 = "{key}-3" + UUID.randomUUID();
+        String value = "{value}" + UUID.randomUUID();
+
+        assertEquals(OK, client.set(key1, value).get());
+        assertEquals(OK, client.set(key2, value).get());
+
+        assertEquals(2, client.touch(new String[] {key1, key2}).get());
+        assertEquals(0, client.touch(new String[] {key3}).get());
+    }
```
Comment (Yury-Fridlyand) [inline java/integTest/src/test/java/glide/SharedCommandTests.java:2787]: Add a test with keys from different slots (should pass)

File: java/client/src/main/java/glide/api/commands/GenericBaseCommands.java (hunk 1 above)
Comment (SanHalacogluImproving) [reply java/client/src/main/java/glide/api/commands/GenericBaseCommands.java:414]: It does not need a multi key note! Works fine with different slots.

File: java/integTest/src/test/java/glide/SharedCommandTests.java (hunk 2 above)
Comment (SanHalacogluImproving) [reply java/integTest/src/test/java/glide/SharedCommandTests.java:2787]: Renamed keys should be good.

File: java/client/src/main/java/glide/api/models/BaseTransaction.java [hunk 3]
```diff
@@ -2351,6 +2352,22 @@ public T objectRefcount(@NonNull String key) {
         return getThis();
     }
 
+    /**
+     * Updates the last access time of specified <code>keys</code> and returns the count of keys that
+     * were updated.
+     *
+     * @apiNote When in cluster mode, the command may route to multiple nodes when <code>keys</code>
+     *     map to different <code>hash slots</code>.
```
Comment (Yury-Fridlyand) [inline java/client/src/main/java/glide/api/models/BaseTransaction.java:2360]: no need in this note here - transaction goes to a single node always

Comment (Yury-Fridlyand) [reply java/client/src/main/java/glide/api/models/BaseTransaction.java:2360]: https://github.com/aws/glide-for-redis/pull/1332#discussion_r1578641144

**Review by Yury-Fridlyand — APPROVED** [review]
LGFM

File: java/client/src/main/java/glide/api/models/BaseTransaction.java (hunk 3 above)
Comment (SanHalacogluImproving) [reply java/client/src/main/java/glide/api/models/BaseTransaction.java]: Fixed.

**Pull request merged**